h2o-py/h2o/frame.py

```diff
@@ -33,7 +33,7 @@
                                     can_use_pandas, quote, normalize_slice, slice_is_normalized, check_frame_id)
 from h2o.utils.typechecks import (assert_is_type, assert_satisfies, Enum, I, is_type, numeric, numpy_ndarray,
                                   numpy_datetime, pandas_dataframe, pandas_timestamp, scipy_sparse, U)
-from h2o.model.model_base import _get_numpy
+from h2o.utils.ext_dependencies import _get_numpy
```

Comment:
this is how this function is used in this file: totally silly!

```python
        np = _get_numpy("H2OFrame.dtype")
        type_map = {
            "enum": np.str, 
            "string": np.str, 
            "int": np.int, 
            "real": np.float, 
            "time": np.str,
            "uuid": np.str
        }
        types_list = list(self.types.values())
        return np.dtype(type_map[types_list[0]])
```
meaning if `numpy` is missing, then `np` is `None` and we get an `AttributeError` 2 lines later.
Instead we should use:
```python
if not can_use_numpy():
    raise ImportError("H2OFrame.dtype function requires numpy to be installed")
import numpy as np
type_map = {
    "enum": np.str, 
    "string": np.str, 
    "int": np.int, 
    "real": np.float, 
    "time": np.str,
    "uuid": np.str
}
types_list = list(self.types.values())
return np.dtype(type_map[types_list[0]])
```

Reply:
Resolved in recent commit.